app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,44 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, :valid_user, :check_expiration, only: %i(edit update)
+
+  def new; end
+
+  def edit; end
+
+  def update
+    if params[:user][:password].empty?
+      @user.errors.add :password, "can't be empty"
+      render "edit"
+    elsif @user.update(user_params)
+      log_in @user
+      flash[:success] = t "password_reset.success"
+      redirect_to @user
+    else
+      render "edit"
+    end
+  end
+
+  private
+
+  def get_user
+    @user = User.find_by(email: params[:email])
```

Comment:
Xử lý trường hợp không tìm thấy user nữa nhé

Reply:
handle user nil ở đây, và ko dùng toán tử an toàn ở dưới cu